internal/testutils/proxy.go

```diff
@@ -16,7 +16,6 @@
  *
  */
 
-// Package testutils contains helpers for transport tests.
 package testutils
```

Comment:
Could you please make this a separate package instead (subdirectory of this one)?  Generally we should split things when possible instead of building one gigantic package named "utils".  See https://google.github.io/styleguide/go/best-practices#util-packages and also https://google.github.io/styleguide/go/best-practices#package-size.  (Re: package-size: of course this isn't exported, so it's not critical, but our testing infrastructure grows over time and we want to make sure it's easy to use for ourselves.  This package is already probably too big.)